batch_correction/batch_correct_combat.Rmd

```diff
@@ -0,0 +1,155 @@
+---
+title: "Batch Correction"
+author: "ALSF CCDL - Candace Savonen"
+output:   
+  html_notebook: 
+    toc: true
+    toc_float: true
+---
+
+*Purpose*: This notebook illustrates how to assess batch effects and how to use
+ComBat to correct for them. 
+
+This method is based on:
+Johnson, WE, Rabinovic, A, and Li, C (2007). Adjusting batch effects in microarray 
+expression data using Empirical Bayes methods. Biostatistics 8(1):118-127.
+
+## 1) Set up libraries
+Install `sva` if you haven't done so before.
+
+```{r eval=FALSE}
+source("https://bioconductor.org/biocLite.R")
+biocLite("sva", suppressUpdates = TRUE)
+```
+
+```{r Upload ComBat functions}
+# Magrittr pipe
+`%>%` <- dplyr::`%>%`
+
+# Load ggplot2 for graphing
+library(ggplot2)
+```
+
+## 2) Import and set up data
+Data downloaded from refine.bio include a metadata tab separated values ("tsv")
+file and a data tsv file. 
+These files were obtained from [refine.bio](https://www.refine.bio/) by entering
+the search term "HEK 293" and then aggregating by `species`.
+
+```{r Import refine.bio data}
+# Put name of data file
+data.filename <- file.path("..", "data", "HOMO_SAPIENS.tsv")
+
+# Unzip the file and direct it to the data folder
+unzip(paste0(data.filename, ".zip"), exdir = file.path("..", "data"))
+
+# Read in data tsv file
+df <- readr::read_tsv(data.filename) %>% 
+  tibble::column_to_rownames('X1') 
+# Make the gene column the rownames so the gene names are out of the way for calculations
+```
+
+Prep metadata to be in proper format for batch info to be used by ComBat
+
+```{r Prep metadata}
+# Metadata file name
+metadata.filename <- file.path("..", "data", "metadata_HOMO_SAPIENS.tsv")
+
+# Read in data tsv file
+metadata <- readr::read_tsv(metadata.filename)
+
+# dataset <- as.factor(stringr::word(colnames(df), sep = ":", 1))
+df <- df[, metadata$refinebio_accession_code]
+
+# Check if the samples are in the same order
+all.equal(metadata$refinebio_accession_code, colnames(df))
+```
+
+The refine.bio search gave us some cells that aren't HEK293 so we will need to 
+do some filtering. 
+Because the metadata obtained from combining multiple datasets are not always 
+able to be harmonized, we are going to have to search for HEK 293 cells in a 
+couple different columns. 
+
+```{r}
+# Keep only things that have evidence that they are HEK cells
+hek293 <- grep("hek|293|hek 293|hek-293", 
+               paste0(metadata$refinebio_cell_line, metadata$`cell line`, metadata$title), 
+               ignore.case = TRUE)
+
+# Only keep these in the data and metadata
+metadata <- metadata[hek293, ] %>% 
+  readr::write_tsv("filtered_metadata_HOMO_SAPIENS.tsv")
+
+# Only keep these in the data and metadata
+df <- df[, hek293] %>% 
+  readr::write_tsv("filtered_HOMO_SAPIENS.tsv")
+
+# Take a look at the information we have for these data
+metadata
```

Comment:
In the notebook setting, preference for using `dplyr::select()` here so we drop things like age which are not informative.

Reply:
May also be useful to include something like `table(refinebio_platform)` so folks get an idea that multiple platforms  (chips) are present in this dataset.